pkg/apis/core/annotation_key_constants.go

```diff
@@ -122,8 +122,21 @@ const (
 	// This annotation is beta-level and is only honored when PodDeletionCost feature is enabled.
 	PodDeletionCost = "controller.kubernetes.io/pod-deletion-cost"
 
-	// AnnotationTopologyAwareHints can be used to enable or disable Topology
-	// Aware Hints for a Service. This may be set to "Auto" or "Disabled". Any
-	// other value is treated as "Disabled".
-	AnnotationTopologyAwareHints = "service.kubernetes.io/topology-aware-hints"
+	// DeprecatedAnnotationTopologyAwareHints can be used to enable or disable
+	// Topology Aware Hints for a Service. This may be set to "Auto" or
+	// "Disabled". Any other value is treated as "Disabled". This annotation has
+	// been deprecated in favor of the `service.kubernetes.io/topology-mode`
+	// annotation. When both `service.kubernetes.io/topology-aware-hints` and
+	// `service.kubernetes.io/topology-mode` annotations are set, the value of
+	// `service.kubernetes.io/topology-mode` has precedence.
```

Comment:
nit: possibly doesn't matter, but it occurred to me that it _might_ be helpful for some readers if we spell it out a bit more that `topology-mode` can still be used like hints if desired:

```suggestion
	// DeprecatedAnnotationTopologyAwareHints can be used to enable or disable
	// Topology Aware Hints for a Service. This may be set to "Auto" or "Disabled".
	// Any other value is treated as "Disabled". This annotation has been
	// deprecated in favor of the `service.kubernetes.io/topology-mode` annotation
	// which also allows "Auto" and "Disabled", but is not limited to those (it's open
	// ended to provide room for experimentation while we pursue configuration for
	// topology via specification). When both
	// `service.kubernetes.io/topology-aware-hints` and
	// `service.kubernetes.io/topology-mode` annotations are set, the value of
	// `service.kubernetes.io/topology-mode` has precedence.
```

Reply:
Updated, thanks!